src/contextmenu.js

```diff
@@ -0,0 +1,472 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Multiple selection context menu.
+ */
+
+import * as Blockly from 'blockly/core';
+import {blockSelection, hasSelectedParent} from './global';
+
+/**
+ * Modification for context menu 'Duplicate' to be available for
+ * multiple blocks.
+ */
+const registerDuplicate = function() {
+  const duplicateOption = {
+    displayText: function() {
+      const selectedBlocksLength = blockSelection.size;
+      if (selectedBlocksLength <= 1) {
+        return Blockly.Msg['DUPLICATE_BLOCK'];
+      } else {
+        return Blockly.Msg['DUPLICATE_BLOCK'] + ' (' +
+        selectedBlocksLength + ')';
+      }
+    },
+    preconditionFn: function(scope) {
+      const block = scope.block;
+      if (!block.isInFlyout && block.isDeletable() && block.isMovable()) {
+        if (block.isDuplicatable()) {
+          return 'enabled';
+        }
+        return 'disabled';
+      }
+      return 'hidden';
+    },
+    callback: function(scope) {
+      const duplicatedBlocks = [];
+      const apply = function(block) {
+        if (block && duplicateOption.preconditionFn({
+          block: block,
+        }) === 'enabled' && !hasSelectedParent(block)) {
+          duplicatedBlocks.push(Blockly.clipboard.duplicate(block));
+        }
+      };
+      Blockly.Events.setGroup(true);
+      if (scope.block && blockSelection.size === 0) {
```

Comment:
```suggestion
      if (scope.block && !blockSelection.size) {
```
Nit: formatting. This works in JavaScript because 0 is false-y. Here & below.